guava/src/com/google/common/primitives/Bytes.java

```diff
@@ -340,9 +335,8 @@ public Byte set(@IndexFor("this") int index, Byte element) {
     }
 
     @Override
-    // array should be @LongerThanEq(value="this", offset = "start")
-    @SuppressWarnings("upperbound:argument.type.incompatible") // custom coll. with size end-start
-    public List<Byte> subList(@IndexOrHigh("this") int fromIndex, @IndexOrHigh("this") int toIndex) {
+    @SuppressWarnings("index") // needs https://github.com/kelloggm/checker-framework/issues/229
+    public List<Byte> subList(@IndexOrHigh("this") @LessThan("#2") int fromIndex, @IndexOrHigh("this") int toIndex) {
```

Comment:
Should be `@LessThan("#2+1")`, because `fromIndex==toIndex` is valid.
`Booleans`, `Floats`, `Integers`, `Longs` and `Shorts` don't have such annotation in this PR.

Reply:
That any of them are annotated is a mistake. This method in the JDK doesn't have a LessThan annotation, and we'd have to suppress the warning anyway even if it did, so there's no reason to add it here.

At some point we should probably change the definition in the JDK, but this pull request isn't the right place. I've removed them.